tests/local/memory-discard.wat

```diff
@@ -0,0 +1,7 @@
+(module
+  (memory i32 16)
+
+  (func
+    (memory.discard (i32.const 0) (i32.const 16))
```

Comment:
This is actually not great since my prototype uses bytes instead of pages, but when I update this test + snapshots again (`BLESS=1 cargo test`) it obliterates most of the snapshots in the repo. Trying to sort that out.